src/progress.rs

```diff
@@ -200,18 +200,29 @@ pub struct SettingsUpdate {
     pub displayed_attributes: UpdateState<BTreeSet<String>>,
     pub stop_words: UpdateState<BTreeSet<String>>,
     pub synonyms: UpdateState<BTreeMap<String, Vec<String>>>,
-    pub attributes_for_faceting: UpdateState<Vec<String>>,
+    pub filterable_attributes: UpdateState<Vec<String>>,
 }
 
 #[derive(Debug, Clone, Deserialize)]
 #[serde(tag = "name")]
 pub enum UpdateType {
     ClearAll,
     Customs,
-    DocumentsAddition { number: usize },
-    DocumentsPartial { number: usize },
-    DocumentsDeletion { number: usize },
-    Settings { settings: Box<SettingsUpdate> },
+    DocumentsAddition {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        number: Option<usize>,
+    },
+    DocumentsPartial {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        number: Option<usize>,
+    },
+    DocumentsDeletion {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        number: Option<usize>,
+    },
+    Settings {
+        settings: SettingsUpdate,
+    },
```

Comment:
Thanks a lot for raising this change, was this documented somewhere?

I have made an issue to know if this is the expected behavior or not: see meilisearch/MeiliSearch/issues/1517